### PR TITLE
fix(opportunity): append ?link_preview=false to minted short URLs + log actionability

### PR DESCRIPTION
**Pull request by yanekyuk**
## Summary

Two related fixes:

1. The MCP-path `acceptUrl` (a short URL `${apiBaseUrl}/c/<code>` minted by the backend's `mintConnectLink` adapter) was missing the `?link_preview=false` hint that the profileUrl picked up in #770. As a result, Telegram still rendered a link preview for the connect link when there was no preview-worthy URL elsewhere in the message. The fix wires the suffix in at the backend composition root (`protocol-init.ts`), where the adapter is configured — the protocol package itself stays unaware of channel rendering concerns.
2. Adds per-card actionability logging so when an agent surfaces N opportunities but only M of them have an `acceptUrl`, you can see in the logs exactly which (`status × viewerRole × viewerApproved`) combination resolved to which `kind` (or `none`).

## Bug Fixes

- **opportunity**: `backend/src/protocol-init.ts` mintConnectLink adapter now returns short URLs of the form `${apiBaseUrl}/c/<code>?link_preview=false`. Treats `?link_preview=false` as backend-side rendering config, mirroring how profileUrls already carry it. The protocol package's `attachActionableLinks` just echoes whatever the adapter returns.

## Observability

- **opportunity.tools**: New info-level `"Opportunity actionability decision"` log in `attachActionableLinks` — per card, surfaces `opportunityId`, `status`, `viewerRole`, `viewerApproved`, and resolved `kind` (or `"none"`). Makes the silent "card returned but not actionable" path visible.
- **tool.factory**: MCP tool-invocation log (`"Tool: <name>"` with `userId`, `networkId`, `query`) promoted from `verbose` to `info` so it's visible at the default log level without needing `LOG_LEVEL=verbose` on Railway.

## Other

- Bumps `@indexnetwork/protocol` 0.30.7 → 0.30.8.

## Test plan

- [x] `bun test packages/protocol/src/opportunity/tests/opportunity.tools.spec.ts` — 40/40 pass; new log line fires in test output
- [x] `tsc --noEmit` clean in `packages/protocol` and `backend`
- [ ] Re-test EdgeClaw: send a message that surfaces multiple opportunities; verify (a) Telegram no longer shows a preview for the connect-link URL, (b) Railway logs show one `Opportunity actionability decision` line per card with the resolved kind

### CODE REVIEW

File: packages/protocol/src/shared/agent/tool.factory.ts [hunk 1]
```diff
@@ -84,7 +84,7 @@ export async function createChatTools(
   }) {
     return tool(
       async (query: z.infer<T>) => {
-        logger.verbose(`Tool: ${opts.name}`, {
+        logger.info(`Tool: ${opts.name}`, {
           context: { userId: resolvedContext.userId, networkId: resolvedContext.networkId },
           query,
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply packages/protocol/src/shared/agent/tool.factory.ts]: Fixed in bb1bcc9b8 — `tool.factory.ts` now wraps `query` with `redactSensitiveFields(...)` before logging, matching the parallel `tool.registry.ts` path. Sensitive fields are scrubbed before the info-level line is emitted.

File: backend/src/protocol-init.ts [hunk 2]
```diff
@@ -58,7 +58,7 @@ export function createDefaultProtocolDeps(): ProtocolDeps {
   ).replace(/\/+$/, '');
   const mintConnectLink: MintConnectLink = async ({ userId, opportunityId, kind, greeting }) => {
     const { code } = await mintConnectLinkSvc({ userId, opportunityId, kind, greeting });
-    return { url: `${apiBaseUrl}/c/${code}` };
+    return { url: `${apiBaseUrl}/c/${code}?link_preview=false` };
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply backend/src/protocol-init.ts]: Fixed in bb1bcc9b8 — extracted `buildConnectShortUrl(apiBaseUrl, code)` into `connect-link.service.ts` and added unit tests in `connect-link.service.spec.ts` that assert the `?link_preview=false` suffix. `protocol-init.ts` delegates to the helper, so a regression that drops the query parameter now fails the new tests.

File: packages/protocol/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@indexnetwork/protocol",
-  "version": "0.30.7",
+  "version": "0.30.8",
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply packages/protocol/package.json:3]: Holding on this — recent PRs in this repo (e.g. #770) omit empty changelog sections rather than padding with placeholders, and the CLAUDE.md guidance reads as describing the section *vocabulary* to use when content exists, not a strict template requiring every section header on every PR. This PR's description covers what changed (Summary, Bug Fixes, Observability, Test Plan); adding empty New Features / Refactors / Documentation sections would be noise. Happy to revisit if the maintainers want to formalize the convention.
